### PR TITLE
[Vision] Update to Xcode 9 Beta 5

### PR DESCRIPTION
**Pull request by dalexsoto**

### CODE REVIEW

File: src/vision.cs [hunk 1]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
 	[DisableDefaultCtor]
 	interface VNTargetedImageRequest {
 
-		[Export ("initWithTargetedCVPixelBuffer:")]
-		IntPtr Constructor (CVPixelBuffer pixelBuffer);
+		[Export ("initWithTargetedCVPixelBuffer:options:")]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer, NSDictionary optionsDict);
```
Comment (spouliot) [inline src/vision.cs]: Unless it's confusing use the same parameters name as Apple, i.e. the `Dict` suffix is not needed, `options` is fine.

File: src/vision.cs [hunk 2]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
-		[Export ("initWithTargetedCVPixelBuffer:completionHandler:")]
-		[DesignatedInitializer]
-		IntPtr Constructor (CVPixelBuffer pixelBuffer, [NullAllowed] VNRequestCompletionHandler completionHandler);
+		[Wrap ("this (pixelBuffer, options?.Dictionary)")]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer, VNImageOptions options);
 
-		[Export ("initWithTargetedCGImage:")]
-		IntPtr Constructor (CGImage image);
+		[Export ("initWithTargetedCVPixelBuffer:options:completionHandler:")]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer, NSDictionary optionsDict, [NullAllowed] VNRequestCompletionHandler completionHandler);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 3]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
-		[Export ("initWithTargetedCGImage:completionHandler:")]
-		[DesignatedInitializer]
-		IntPtr Constructor (CGImage image, [NullAllowed] VNRequestCompletionHandler completionHandler);
+		[Wrap ("this (pixelBuffer, options?.Dictionary, completionHandler)")]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer, VNImageOptions options, VNRequestCompletionHandler completionHandler);
 
-		[Export ("initWithTargetedCIImage:")]
-		IntPtr Constructor (CIImage image);
+		[Export ("initWithTargetedCVPixelBuffer:orientation:options:")]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer, CGImagePropertyOrientation orientation, NSDictionary optionsDict);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 4]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
-		[Export ("initWithTargetedCIImage:completionHandler:")]
-		[DesignatedInitializer]
-		IntPtr Constructor (CIImage image, [NullAllowed] VNRequestCompletionHandler completionHandler);
+		[Wrap ("this (pixelBuffer, orientation, options?.Dictionary)")]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer, CGImagePropertyOrientation orientation, VNImageOptions options);
 
-		[Export ("initWithTargetedImageURL:")]
-		IntPtr Constructor (NSUrl imageUrl);
+		[Export ("initWithTargetedCVPixelBuffer:orientation:options:completionHandler:")]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer, CGImagePropertyOrientation orientation, NSDictionary optionsDict, [NullAllowed] VNRequestCompletionHandler completionHandler);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 5]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
-		[Export ("initWithTargetedImageURL:completionHandler:")]
-		[DesignatedInitializer]
-		IntPtr Constructor (NSUrl imageUrl, [NullAllowed] VNRequestCompletionHandler completionHandler);
+		[Wrap ("this (pixelBuffer, orientation, options?.Dictionary, completionHandler)")]
+		IntPtr Constructor (CVPixelBuffer pixelBuffer, CGImagePropertyOrientation orientation, VNImageOptions options, VNRequestCompletionHandler completionHandler);
 
-		[Export ("initWithTargetedImageData:")]
-		IntPtr Constructor (NSData imageData);
+		[Export ("initWithTargetedCGImage:options:")]
+		IntPtr Constructor (CGImage cgImage, NSDictionary optionsDict);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 6]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
 
-		[Export ("initWithTargetedImageData:completionHandler:")]
-		[DesignatedInitializer]
-		IntPtr Constructor (NSData imageData, [NullAllowed] VNRequestCompletionHandler completionHandler);
+		[Wrap ("this (cgImage, options?.Dictionary)")]
+		IntPtr Constructor (CGImage cgImage, VNImageOptions options);
+
+		[Export ("initWithTargetedCGImage:options:completionHandler:")]
+		IntPtr Constructor (CGImage cgImage, NSDictionary optionsDict, [NullAllowed] VNRequestCompletionHandler completionHandler);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 7]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (cgImage, options?.Dictionary, completionHandler)")]
+		IntPtr Constructor (CGImage cgImage, VNImageOptions options, VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedCGImage:orientation:options:")]
+		IntPtr Constructor (CGImage cgImage, CGImagePropertyOrientation orientation, NSDictionary optionsDict);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 8]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (cgImage, orientation, options?.Dictionary)")]
+		IntPtr Constructor (CGImage cgImage, CGImagePropertyOrientation orientation, VNImageOptions options);
+
+		[Export ("initWithTargetedCGImage:orientation:options:completionHandler:")]
+		IntPtr Constructor (CGImage cgImage, CGImagePropertyOrientation orientation, NSDictionary optionsDict, [NullAllowed] VNRequestCompletionHandler completionHandler);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 9]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (cgImage, orientation, options?.Dictionary, completionHandler)")]
+		IntPtr Constructor (CGImage cgImage, CGImagePropertyOrientation orientation, VNImageOptions options, VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedCIImage:options:")]
+		IntPtr Constructor (CIImage ciImage, NSDictionary optionsDict);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 10]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (ciImage, options?.Dictionary)")]
+		IntPtr Constructor (CIImage ciImage, VNImageOptions options);
+
+		[Export ("initWithTargetedCIImage:options:completionHandler:")]
+		IntPtr Constructor (CIImage ciImage, NSDictionary optionsDict, [NullAllowed] VNRequestCompletionHandler completionHandler);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 11]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (ciImage, options?.Dictionary, completionHandler)")]
+		IntPtr Constructor (CIImage ciImage, VNImageOptions options, VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedCIImage:orientation:options:")]
+		IntPtr Constructor (CIImage ciImage, CGImagePropertyOrientation orientation, NSDictionary optionsDict);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 12]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (ciImage, orientation, options?.Dictionary)")]
+		IntPtr Constructor (CIImage ciImage, CGImagePropertyOrientation orientation, VNImageOptions options);
+
+		[Export ("initWithTargetedCIImage:orientation:options:completionHandler:")]
+		IntPtr Constructor (CIImage ciImage, CGImagePropertyOrientation orientation, NSDictionary optionsDict, [NullAllowed] VNRequestCompletionHandler completionHandler);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 13]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (ciImage, orientation, options?.Dictionary, completionHandler)")]
+		IntPtr Constructor (CIImage ciImage, CGImagePropertyOrientation orientation, VNImageOptions options, VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedImageURL:options:")]
+		IntPtr Constructor (NSUrl imageUrl, NSDictionary optionsDict);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 14]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (imageUrl, options?.Dictionary)")]
+		IntPtr Constructor (NSUrl imageUrl, VNImageOptions options);
+
+		[Export ("initWithTargetedImageURL:options:completionHandler:")]
+		IntPtr Constructor (NSUrl imageUrl, NSDictionary optionsDict, [NullAllowed] VNRequestCompletionHandler completionHandler);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 15]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (imageUrl, options?.Dictionary, completionHandler)")]
+		IntPtr Constructor (NSUrl imageUrl, VNImageOptions options, VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedImageURL:orientation:options:")]
+		IntPtr Constructor (NSUrl imageUrl, CGImagePropertyOrientation orientation, NSDictionary optionsDict);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 16]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (imageUrl, orientation, options?.Dictionary)")]
+		IntPtr Constructor (NSUrl imageUrl, CGImagePropertyOrientation orientation, VNImageOptions options);
+
+		[Export ("initWithTargetedImageURL:orientation:options:completionHandler:")]
+		IntPtr Constructor (NSUrl imageUrl, CGImagePropertyOrientation orientation, NSDictionary optionsDict, [NullAllowed] VNRequestCompletionHandler completionHandler);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 17]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (imageUrl, orientation, options?.Dictionary, completionHandler)")]
+		IntPtr Constructor (NSUrl imageUrl, CGImagePropertyOrientation orientation, VNImageOptions options, VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedImageData:options:")]
+		IntPtr Constructor (NSData imageData, NSDictionary optionsDict);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 18]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (imageData, options?.Dictionary)")]
+		IntPtr Constructor (NSData imageData, VNImageOptions options);
+
+		[Export ("initWithTargetedImageData:options:completionHandler:")]
+		IntPtr Constructor (NSData imageData, NSDictionary optionsDict, [NullAllowed] VNRequestCompletionHandler completionHandler);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 19]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (imageData, options?.Dictionary, completionHandler)")]
+		IntPtr Constructor (NSData imageData, VNImageOptions options, VNRequestCompletionHandler completionHandler);
+
+		[Export ("initWithTargetedImageData:orientation:options:")]
+		IntPtr Constructor (NSData imageData, CGImagePropertyOrientation orientation, NSDictionary optionsDict);
```
Comment (spouliot) [inline src/vision.cs]: same

File: src/vision.cs [hunk 20]
```diff
@@ -804,40 +814,125 @@ interface VNSequenceRequestHandler {
...
+
+		[Wrap ("this (imageData, orientation, options?.Dictionary)")]
+		IntPtr Constructor (NSData imageData, CGImagePropertyOrientation orientation, VNImageOptions options);
+
+		[Export ("initWithTargetedImageData:orientation:options:completionHandler:")]
+		IntPtr Constructor (NSData imageData, CGImagePropertyOrientation orientation, NSDictionary optionsDict, [NullAllowed] VNRequestCompletionHandler completionHandler);
```
Comment (spouliot) [inline src/vision.cs]: same

**Comment by spouliot** [comment]
@dalexsoto it's rather minor and I prefer to have the new API in the next preview. Please fix those names in a separate PR, thanks!

**Comment by dalexsoto** [comment]
I did this because then using named parameters will be a lil painful, you will need to cast

**Pull request merged**